caluma/caluma_core/tests/test_health_checks.py

```diff
@@ -0,0 +1,542 @@
+import pytest
+
+from .. import models
+from .fake_model import get_fake_model
+from .fake_model import define_fake_model
+from ..health_checks import check_migrations
+from django.db.models import CharField
+from django.test import Client
+import django.db
+from django.db import connections
+from django.db.backends.signals import connection_created
+from django.urls import reverse, resolve
+from django.conf import settings
+import time
+from urllib3.response import HTTPResponse
+from django.test import override_settings
+import xml
+
+from caluma.caluma_form import storage_clients
+from django.db.migrations.executor import MigrationExecutor
+
+from minio import Minio
+from minio.definitions import Object as MinioStatObject
+from minio import ResponseError
+
+from django.core import management
+
+import sys
+
+from django.db.migrations import Migration
+from watchman import settings as watchman_settings
+
+
+@pytest.mark.parametrize("working_db", [True, False])
+def test_db_connection(request, client, snapshot, minio_mock_working, working_db):
+    """
+    Test /healthz/ endpoint response when database connection is
+    available or unavailable.
+    Conditionally provides database access through the 'db' fixture.
+    This affects the watchman check for databases and the db models
+    check and prevents access to the database.
+    """
+
+    # use database fixture if test in working state
+    if working_db:
```

Comment:
This test is basically split in two, with only two lines of common code. I think it would be better to remove the parametrization and turn it into two separate tests. This may also solve your problem with the fixtures and the migrations

Reply:
Split all tests, because this was the case for all of them. Removed duplicate tests, usually the ones that checked that everything was working with a `200` response. 
It _did_ solve the problem with the fixtures and the migrations.

Reply:
The db connection error test now uses a db fixture that cannot connect to the database (by modifying the database host setting). Instead of not requesting the `db` fixture.